lib/es6-promise/polyfill.js

```diff
@@ -2,7 +2,7 @@
 import Promise from './promise';
 
 export default function polyfill() {
-  var local = (typeof global !== 'undefined') ? global : self;
+  var local = Function('return this;')();
```

Comment:
How about the following?

```
var local = (typeof global !== 'undefined') ? global :
            (typeof window !== 'undefined' && window.document) ? window : Function('return this;')();
```

It must work on eval-disabled browsers, and also supports generic environment.